src/modules/rewards/hooks/useVerifyRewards.tsx

```diff
@@ -0,0 +1,95 @@
+// react and other libraries
+import { useEffect, useState } from 'react';
+
+// third party libraries
+import { useSelector } from 'react-redux';
+
+// helpers
+import { generateVerificationProof } from '../utils/generateVerificationProof';
+import { useClaimRewardsActivity } from 'queries';
+
+// types
+import { UserStoreType } from 'types';
+
+export type UseVerifyRewardsParams = {
+  activityTypeId: string;
+  setErrorMessage: (errorMessage: string) => void;
+  refetchActivity: () => void;
+};
+
+const useVerifyRewards = ({ activityTypeId, setErrorMessage, refetchActivity }: UseVerifyRewardsParams) => {
+  const [verifyingRewards, setVerifyingRewards] = useState(false);
+  const [rewardsActivityStatus, setRewardsActivityStatus] = useState<string | null>(null);
+  const { userPushSDKInstance } = useSelector((state: UserStoreType) => state.user);
+  const [updatedId, setUpdatedId] = useState<string | null>(null);
+
+  useEffect(() => {
+    setErrorMessage('');
+  }, [setErrorMessage]);
+
+  const handleRewardsVerification = (userId: string) => {
+    setUpdatedId(userId);
+    setVerifyingRewards(true);
+    handleVerify(userId);
+  };
+
+  const { mutate: claimRewardsActivity } = useClaimRewardsActivity({
+    userId: updatedId as string,
+    activityTypeId,
+  });
+
+  const handleVerify = async (userId: string | null) => {
+    setErrorMessage('');
+
+    const data = {};
+
+    const verificationProof = await generateVerificationProof(data, userPushSDKInstance);
```

Comment:
```suggestion

    const verificationProof = await generateVerificationProof({}, userPushSDKInstance);
```

is this expected? passing empty object.

Reply:
Verified this with the BE team, for other rewards apart from Twitter and discord. We pass just the empty data object